### DISCUSSION TITLE
{scroll: false} not working

### DISCUSSION THREAD

**Discussion by jerrywu28** [vercel/next.js#35072] [Help]
Hello all! I have a page with a list of items where the query params are set with `router.replace` whenever a list item is clicked. Unfortunately, I'm running into an issue where the page scrolls to the top every time a list item is clicked. I'm using the `replace` method as something like:

`router.replace(url, undefined, { scroll: false }`

This doesn't prevent the scroll-to-top from happening. I've also tried adding shallow scrolling to the options and set undefined as the same argument as the url, but to no avail. Anyone have any ideas for how to resolve this? Thanks in advance!

**Comment by chemicalkosek** [comment]
You need to use `window.history.replaceState()` if you just want to change the query param without changing page.

**Reply by jerrywu28** [reply]
> You need to use `window.history.replaceState()` if you just want to change the query param without changing page.

Thanks for replying! Should passing in the option of `{ shallow: true }` do the same thing?

**Reply by chemicalkosek** [reply]
No. `shallow` means that getStaticProps and getServerSideProps won't be called while routing.

**Reply by jerrywu28** [reply]
I see. Thanks for clarifying - I'll try using the replaceState method you mentioned.

**Reply by christiankaindl** [reply]
It should also work with Next.js router, without `window.history.replaceState()`

From the docs:
> To prevent scrolling to the top / hash `scroll={false}` can be added to `Link`

See https://nextjs.org/docs/api-reference/next/link#disable-scrolling-to-the-top-of-the-page

**Reply by christiankaindl** [reply]
Using the Next.js router with `scroll=false` doesn't work for me on v12.3...

Related: https://github.com/vercel/next.js/discussions/13804
